fulp_modules/lisa/joel_gun.dm

```diff
@@ -0,0 +1,195 @@
+			 "<span class='userdanger'>You spin [src] around!</span>",\
+			 "<span class='hear'>You hear metal clanking...</span>")
+			for(var/mob/living/L in viewers(7, user))
+				L.face_atom(user)
+				L.do_alert_animation()
+			playsound(src, 'fulp_modules/lisa/Sounds/gunreveal.ogg', 20, FALSE, -5)
+			var/mob/living/carbon/C = user
+			used_ability = TRUE
+			addtimer(CALLBACK(src, .proc/clear_cooldown), 6 SECONDS)
+			C.adjustStaminaLoss(-10)
+			return
+		else /// Otherwise, just toss it.
+			user.visible_message("<span class='danger'>[user.name] throws [src] around!</span>",\
+			 "<span class='userdanger'>You toss [src] around!</span>",\
+			 "<span class='hear'>You hear metal clanking...</span>")
+			for(var/mob/living/L in viewers(7, user))
+				L.face_atom(user)
+				L.do_alert_animation()
+			playsound(src, 'fulp_modules/lisa/Sounds/guntoss.ogg', 50, FALSE, -5)
+			var/mob/living/carbon/C = user
+			used_ability = TRUE
+			addtimer(CALLBACK(src, .proc/clear_cooldown), 6 SECONDS)
+			C.adjustStaminaLoss(-10)
+			C.drop_all_held_items()
+			return
+	..()
+
+/// Gun reveal + Mind Games
+/obj/item/gun/ballistic/revolver/joel/AltClick(mob/user)
+	if(!used_ability)
+		if(HAS_TRAIT(user, TRAIT_GUNFLIP))
+			user.visible_message("<span class='danger'>[user.name] opens their holster and pulls out [src]!</span>",\
+			 "<span class='userdanger'>You reveal [src] from your holster!</span>",\
+			 "<span class='hear'>You hear metal clanking...</span>")
+			for(var/mob/living/carbon/human/L in viewers(7, user))
+				if(prob(15) && !(L == user) && !(L.stat))
+					to_chat(L, "<span class='warning'>Seeing [src] revealed in such a manner disgusts you!</span>")
+					if(!L.mind || !istype(L.mind.martial_art, /datum/martial_art/velvetfu))
+						L.vomit(0, FALSE, FALSE, 3, TRUE, harm = FALSE)
+			playsound(src, 'fulp_modules/lisa/Sounds/gunreveal.ogg', 20, FALSE, -5)
+			used_ability = TRUE
+			addtimer(CALLBACK(src, .proc/clear_cooldown), 10 SECONDS)
+			return
+		else /// No holster? Use mind games instead
+			user.visible_message("<span class='danger'>[user.name] stares deeply around [user.p_them()]selves!</span>",\
+			 "<span class='userdanger'>You stare directly into the souls of those around you.</span>",\
+			 "<span class='hear'>You hear metal clanking...</span>")
+			for(var/mob/living/H in viewers(1, user))
+				if(prob(75) && !(H == user) && !(H.stat))
+					to_chat(H, "<span class='warning'>You feel terrifyingly spooked by [user.name]!</span>")
+					if(!H.mind || !istype(H.mind.martial_art, /datum/martial_art/velvetfu))
+						var/throwtarget = get_edge_target_turf(user, get_dir(user, get_step_away(H, user)))
+						H.throw_at(throwtarget, 3, 2)
+			used_ability = TRUE
+			addtimer(CALLBACK(src, .proc/clear_cooldown), 10 SECONDS)
+			return
+	else
+		..()
+		spin()
+
+/datum/voice_of_god_command/repulse/execute(list/listeners, mob/living/user, power_multiplier = 1, message)
```

Comment:
???????????????????

Reply:
I copy pasted it to use it for reference (and steal) and forgot to delete afterwards.